.github/workflows/tests.yaml

```diff
@@ -58,7 +60,11 @@ jobs:
         uses: actions/checkout@v2
         with:
           submodules: true
-      - name: Install Dependencies
+      - name: Install Dependencies (without optional)
+        if: ${{ matrix.optional-deps == '' }}
         run: composer install
+      - name: Install Dependencies (with optional)
+        if: ${{ matrix.optional-deps != '' }}
+        run: composer require ${{ matrix.optional-deps }}
```

Comment:
The matrix will allow us to run these tests twice:

 - Once without `cache/filesystem-adapter` (and therefore using `psr/simple-cache` 3.x)
 - Once with `cache/filesystem-adapter` (and therefore using `psr/simple-cache` 1.x)

Thus giving us confidence that everything works as expected.